File: examples/basic-capture.py
"""basic-capture.py — Webshot SDK quickstart in Python.
Install: pip install webshot
Run:     python basic-capture.py
"""
import sys
from webshot import WebshotClient, WebshotThrottledError, WebshotError

with WebshotClient() as client:
    try:
        shot = client.capture("https://example.com", format="png", mode="desktop_full")
        shot.write("example.png")
        print(f"✓ saved example.png ({len(shot.bytes)} bytes)")
    except WebshotThrottledError as e:
        print(f"Throttled — retry in {e.retry_after}s. Email {e.contact} for higher limits.")
        sys.exit(1)
    except WebshotError as e:
        print(f"Capture failed: {e}")
        sys.exit(1)

File: examples/basic-capture.js
// basic-capture.js — Webshot SDK quickstart in Node.
// Install: npm install @tuxxin/webshot
// Run:     node basic-capture.js
import { WebshotClient, WebshotThrottledError } from '@tuxxin/webshot';
import { writeFile } from 'node:fs/promises';

const client = new WebshotClient();

try {
  const shot = await client.capture({
    url:    'https://example.com',
    format: 'png',
    mode:   'desktop_full',
  });
  await writeFile('example.png', shot.bytes);
  console.log(`✓ saved example.png (${shot.bytes.length} bytes)`);
} catch (err) {
  if (err instanceof WebshotThrottledError) {
    console.error(`Throttled — retry in ${err.retryAfter}s. Email ${err.contact} for higher limits.`);
  } else {
    console.error(`Capture failed: ${err.message}`);
  }
  process.exit(1);
}
